front/components/agent_builder/triggers/AgentBuilderTriggersBlock.tsx

```diff
@@ -0,0 +1,174 @@
+        <TimeIcon className="h-4 w-4 text-foreground dark:text-foreground-night" />
+      );
+    default:
+      return null;
+  }
+}
+
+interface TriggerCardProps {
+  trigger: AgentBuilderTriggerType;
+  onRemove: () => void;
+  onEdit?: () => void;
+}
+
+function TriggerCard({ trigger, onRemove, onEdit }: TriggerCardProps) {
+  return (
+    <Card
+      variant="primary"
+      className="h-28"
+      onClick={onEdit}
+      action={
+        <CardActionButton
+          size="mini"
+          icon={XMarkIcon}
+          onClick={(e: Event) => {
+            e.stopPropagation();
+            onRemove();
+          }}
+        />
+      }
+    >
+      <div className="flex w-full flex-col gap-2 text-sm">
+        <div className="flex w-full items-center gap-2 font-medium text-foreground dark:text-foreground-night">
+          {getIcon(trigger.kind)}
+          <span className="truncate">{trigger.name}</span>
+        </div>
+
+        <div className="text-muted-foreground dark:text-muted-foreground-night">
+          <span className="line-clamp-2 break-words">
+            {trigger.description}
+          </span>
+        </div>
+      </div>
+    </Card>
+  );
+}
+
+interface AgentBuilderTriggersBlockProps {
+  isTriggersLoading?: boolean;
+}
+
+export function AgentBuilderTriggersBlock({
+  isTriggersLoading,
+}: AgentBuilderTriggersBlockProps) {
+  const {
+    fields: triggers,
+    remove,
+    append,
+    update,
+  } = useFieldArray<AgentBuilderFormData, "triggers">({
+    name: "triggers",
+  });
+
+  const sendNotification = useSendNotification();
+  const [editingTrigger, setEditingTrigger] = useState<{
+    trigger: AgentBuilderTriggerType;
+    index: number;
+  } | null>(null);
+  const [isCreateModalOpen, setIsCreateModalOpen] = useState(false);
+
+  const handleCreateTrigger = () => {
+    setIsCreateModalOpen(true);
+  };
+
+  const handleTriggerEdit = (
+    trigger: AgentBuilderTriggerType,
+    index: number
+  ) => {
+    setEditingTrigger({ trigger, index });
+  };
+
+  const handleCloseModal = () => {
+    setEditingTrigger(null);
+    setIsCreateModalOpen(false);
+  };
+
+  const handleTriggerSave = (trigger: AgentBuilderTriggerType) => {
+    if (editingTrigger) {
+      update(editingTrigger.index, trigger);
+    } else {
+      append(trigger);
+    }
+    handleCloseModal();
+  };
+
+  const handleTriggerRemove = (
+    trigger: AgentBuilderTriggerType,
+    index: number
+  ) => {
+    remove(index);
+    sendNotification({
+      type: "success",
+      title: `Successfully removed ${trigger.name}`,
+      description: `Trigger "${trigger.name}" will be removed when you save the agent.`,
+    });
+  };
+
+  return (
+    <AgentBuilderSectionContainer
+      title="Triggers"
+      description="Triggers agent execution based on events."
+      headerActions={
+        <Button
+          label="Add Schedule"
+          variant="primary"
+          icon={ClockIcon}
+          onClick={handleCreateTrigger}
```

Comment:
You're gonna need to a `type="button"` to make sure we don't submit the form on click